.github/workflows/test.yaml

```diff
@@ -69,7 +69,19 @@ jobs:
       sha: ${{ inputs.sha }}
       script: "ci/test_python_scikit_learn_tests.sh"
       # Select amd64 and one job per major CUDA version with the latest CUDA and Python versions
-      matrix_filter: map(select(.ARCH == "amd64")) | group_by(.CUDA_VER|split(".")|map(tonumber)|.[0]) | map(max_by([.CUDA_VER,.PY_VER]|map(split(".")|map(tonumber))))
+      # Add an "intermediate" entry based on the "oldest" entry
+      matrix_filter: '(
+        map(select(.ARCH == "amd64"))
+        | group_by(.CUDA_VER|split(".")|map(tonumber)|.[0])
+        | map(max_by([.CUDA_VER,.PY_VER]|map(split(".")|map(tonumber))))
+        )
+      +
+      (
+        map(select(.ARCH == "amd64"))
+        | map(select(.DEPENDENCIES == "oldest"))
+        | sort_by(.PY_VER)
+        | map(.DEPENDENCIES = "intermediate")
+      )'
```

Comment:
Could we copy this to also apply to `pr.yaml`? It shouldn't up our CI runtime, and would help ensure we don't merge PRs that will fail on sklearn 1.5.2.

Reply:
Yes we can. Just to double check: it will up the CI run time, but probably not by a lot. We have to run the scikit-learn tests one extra time.

One chocolate bar says that we will already discover a failing test if I add it now. Let's find out

Reply:
I think we initially discussed only running those in nightlies, but considering that these tests run in parallel (and thus should usually not have any significant effect on CI walltime), I agree, that it's probably a good idea to expand this to PR CI.

Reply:
I was wrong, all tests passed :-o

Merging